node/rpc/server.go

```diff
@@ -0,0 +1,214 @@
+package rpcserver
+
+import (
+	"context"
+	"errors"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/NethermindEth/juno/jsonrpc"
+	"github.com/NethermindEth/juno/rpc"
+	"github.com/NethermindEth/juno/service"
+	"github.com/NethermindEth/juno/utils"
+	"github.com/NethermindEth/juno/validator"
+	"github.com/sourcegraph/conc"
+)
+
+type Server struct {
+	listener net.Listener
+	mux      *http.ServeMux
+}
+
+var _ service.Service = (*Server)(nil)
+
+func New(listener net.Listener, rpcHandler *rpc.Handler, log utils.SimpleLogger) (*Server, error) {
```

Comment:
Now that we are moving towards a single HTTP server, we can just pass a port here instead of a listener, no?

Reply:
Explicit ports cause flaky tests.

Reply:
Can't we just pass 0 as port in tests?